File: 2st/Riyeon/다항식 더하기.py
def solution(polynomial):
    li = polynomial.split(' + ')
    x, n = 0, 0
    
    for i in li:
        if 'x' in i:
            x += 1 if len(i)==1 else int(i[:-1])
        else:
            n += int(i)
    
    if x == 0 and n == 0:
        return '0'
    elif x == 0:
        return str(n)
    
    if x == 1:
        x = ''
    if n == 0:
        return str(x)+'x'
    else:
        return str(x)+'x + '+str(n)

File: 4st/Eunseong/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        int x = 0, num = 0;
        String pol[] = polynomial.split(" \\+ ");

        for (int i = 0; i < pol.length; i++) {
            if (pol[i].contains("x")) {
                if (pol[i].equals("x")) x++;
                else x += Integer.valueOf(pol[i].replace("x", ""));
            } else {
                num += Integer.valueOf(pol[i]);
            }
        }

        if (x == 0) {
            if (num == 0) return answer;
            else return answer += num;
        } else {
            if (x == 1) answer += "x";
            else answer += x + "x";
            if (num == 0) return answer;
            else return answer += " + " + num;
        }
    }
}